### ISSUE TITLE
SlackRTM prints slack messages both in Hangouts and Slack

### ISSUE THREAD

**Issue by MarkNBroadhead** [hangoutsbot/hangoutsbot#887]
## Description
SlackRTM prints Slack messages both in Hangouts _and_ Slack. 
## Output of `/bot version`
2.8.0
## Expected behaviour
When synchronizing a hangouts channel and Slack channel, SlackRTM plugin and hangoutsbot should print out one message in Hangouts for each Slack message entered and one Slack message for each Hangouts message entered.
## Actual behaviour
Behavior from Hangouts to Slack is normal, however, when entering a message from a Slack channel, the bot outputs the message in Hangouts and then reads what it wrote in Hangouts and prints it again in Slack.
## Steps to reproduce:
  1. Install, run once to generate dummy configuration
  2. Enter SlackRTM configuration in config.json to mirror Slack channel
  3. Add bot to hangouts channel, 
  4. in hangouts 
```
/bot iamspartacus
/bot slack_syncto <Slack Config Alias> <Slack Room Name>
```
  5. Type in that slack room.
  6. Profit!
Note... I do also have the XKCD and catfact plugins enabled in the logs. This issue occurs without these plugins enabled.

## Log output
Sanitized log output follows. Please note lines 50-53. It is having an issue parsing image data. My bot does not have an image for Hangouts. This also appears to be where the bug can be resolved, in slackrtm.py, line 1151, in handle_ho_message. It appears SlackRTM is not recognizing the message is coming from its self and is handling it like any other message. I don't see how this bug is not affecting _everyone_ using this plugin. Regardless, I will submit a fix.
```
2017-09-23 12:41:37 INFO config: <user-dir>/hangupsbot/config.json read
2017-09-23 12:41:37 INFO root: memory = <user-dir>/hangupsbot/memory.json, failsafe = 3, delay = 1
2017-09-23 12:41:37 INFO config: <user-dir>/hangupsbot/memory.json read
2017-09-23 12:41:37 INFO requests.packages.urllib3.connectionpool: Starting new HTTPS connection (1): accounts.google.com
2017-09-23 12:41:38 INFO requests.packages.urllib3.connectionpool: Starting new HTTPS connection (1): www.google.com
2017-09-23 12:41:41 INFO tagging: refreshed
2017-09-23 12:41:41 INFO monkeypatch.otr_support: removeuser replaced with otr_monkeypatch_removeuser
2017-09-23 12:41:41 INFO monkeypatch.otr_support: adduser replaced with otr_monkeypatched_adduser
2017-09-23 12:41:41 INFO plugins: otr_support - no commands
2017-09-23 12:41:41 INFO permamem: loading 6 conversations from memory
2017-09-23 12:41:41 INFO permamem: loading 13 users from hangups
2017-09-23 12:41:41 INFO permamem: loading 6 conversations from hangups
2017-09-23 12:41:41 WARNING permamem: unknown users returned from SlackChannel1 (Ugzls-arx7BmDEocPlh4AaABAQ): ['108966366470725243163']
2017-09-23 12:41:41 WARNING permamem: unknown users returned from SlackChannel2 (UgxptTiMvT9skho3kjx4AaABAQ): ['108966366470725243163']
2017-09-23 12:41:42 WARNING permamem: unknown users returned from SlackChannel3 (Ugzdbxr1ZTWzlDA46qd4AaABAQ): ['108966366470725243163']
2017-09-23 12:41:42 WARNING permamem: unknown users returned from HangoutsChannel1 (UgxeZwV7hj08NOKa7s14AaABAQ): ['102203488014433429067', '108831598871147236385', '117948978379194723306', '108966366470725243163']
2017-09-23 12:41:42 INFO permamem: total conversations: 6
2017-09-23 12:41:42 INFO permamem: total users: 14 cached: 13 definitive (at start): 9
2017-09-23 12:41:42 INFO plugins: plugincontrol - *addplugin, *getplugins, *plugininfo, *pluginload, *pluginreload, *pluginunload, *removeplugin
2017-09-23 12:41:42 INFO plugins: basic - help, *locale, optout, ping, *resourcememory, version
2017-09-23 12:41:42 INFO plugins: tagging - *tagdel, *tagindexdump, *tagscommand, *tagsconv, *tagset, *tagspurge, *tagsuser, *tagsuserlist
2017-09-23 12:41:42 INFO plugins: permamem - *dumpconv, *dumpunknownusers, *makeallusersindefinite, *refreshusermemory, *removeconvrecord, *resetunknownusers
2017-09-23 12:41:42 INFO plugins: convid - *convecho, *convfilter, *convleave, *convrename, *convusers
2017-09-23 12:41:42 INFO plugins: loggertochat - *logconfig, *lograise
2017-09-23 12:41:42 INFO plugins: excluded 52: ['_cam_mail_intercept', '_DEPRECATED_simplytranslate', '_DEPRECATED_syncrooms_autotranslate', '_example', '_example.example_memory', '_example.example_printrenames', '_example.example_watchmembers', '_example.example_webbridge', '_unittest', '_unittest.unittest_geticon', '_unittest.unittest_memory', '_unittest.unittest_reprocessor', '_unittest.unittest_shutup', '_unittest.unittest_statusevents', 'api', 'botaddnotif', 'botalive', 'chance', 'chatlogger', 'cleverbot', 'convtools', 'convtools_invitations', 'easteregg', 'forecast', 'forwarding', 'foursquare', 'humor_hangoutcalls', 'image', 'image.image_linker_reddit', 'image.image_links', 'image.image_memegenerator', 'image.image_screenshot', 'lookup', 'lottery', 'metar', 'monitoradds', 'namelock', 'restrictedadd', 'showme', 'simplewikipedia', 'slack', 'spawn', 'spotify', 'syncrooms', 'syncrooms_config', 'telesync', 'tldr', 'twitter', 'urbandict', 'webbridge_hubot', 'webbridge_telegram', 'wolframalpha']
2017-09-23 12:41:42 INFO plugins: included 11: ['botaliases', 'default', 'mentions', 'subscribe', 'autoreply', 'dnd', 'starter', 'remind', 'xkcd', 'catfact', 'slackrtm']
2017-09-23 12:41:42 INFO plugins.botaliases: aliases: ['/bot', '/chat', '/101378795222110947474']
2017-09-23 12:41:42 INFO plugins: botaliases - botalias
2017-09-23 12:41:42 INFO plugins: default - *broadcast, *config, echo, *hangouts, *leave, *quit, *reload, *rename, *user, *users, *whereami, whoami
2017-09-23 12:41:42 INFO plugins: mentions - bemorespecific, *mention, pushbulletapi, setnickname
2017-09-23 12:41:42 INFO plugins: subscribe - subscribe, *testsubscribe, unsubscribe
2017-09-23 12:41:42 INFO plugins: autoreply - *autoreply
2017-09-23 12:41:42 INFO plugins: dnd - dnd
2017-09-23 12:41:42 INFO plugins: starter - *files, iamspartacus, *test_one2one_message, verifyme
2017-09-23 12:41:42 INFO plugins: remind - remindall, remindme
2017-09-23 12:41:42 INFO plugins: xkcd - xkcd
2017-09-23 12:41:42 INFO plugins: catfact - catfact
2017-09-23 12:41:42 INFO plugins.slackrtm: 1 sink thread(s) started
2017-09-23 12:41:42 INFO plugins: slackrtm - *slack_channels, *slack_disconnect, *slack_listsyncs, *slack_sethotag, *slack_setimageupload, *slack_setslacktag, *slack_setsyncjoinmsgs, *slack_showslackrealnames, *slack_syncto, *slack_users, *slacks
2017-09-23 12:41:42 INFO root: bot initialised
2017-09-23 12:41:42 INFO requests.packages.urllib3.connectionpool: Starting new HTTPS connection (1): slack.com
2017-09-23 12:41:43 INFO plugins.slackrtm: started RTM connection for SlackRTM thread <SlackRTMThread(SlackRTM:EESlack, started daemon 123145448394752)>
2017-09-23 12:41:43 INFO root: list_conversations: 6 from permamem, 6 from hangups - discrepancies: none
2017-09-23 12:41:49 INFO event: eid/dt: 8N2BWVyiOFu8XriZdG908o/2017-09-23 12:41:48
2017-09-23 12:41:49 INFO event: cid/cn: UgxeZwV7hj08NOKa7s14AaABAQ/HangoutsChannel1
2017-09-23 12:41:49 INFO event: c/g/un: 101378795222110947474/101378795222110947474/chat sink
2017-09-23 12:41:49 INFO event: len/tx: 48/marknbroadhead (GroupName): test from slack 2
2017-09-23 12:41:49 WARNING permamem: unknown users returned from HangoutsChannel1 (UgxeZwV7hj08NOKa7s14AaABAQ): ['102203488014433429067', '108831598871147236385', '117948978379194723306', '108966366470725243163']
2017-09-23 12:41:49 ERROR plugins.slackrtm: error while getting user from bot: must be str, not NoneType
Traceback (most recent call last):
  File "/Users/mbroadhead/MyCodeDir/hangoutsbot/hangupsbot/plugins/slackrtm.py", line 1151, in handle_ho_message
    photo_url = "http:" + self.bot._user_list.get_user(event.user_id).photo_url
TypeError: must be str, not NoneType
2017-09-23 12:41:49 INFO requests.packages.urllib3.connectionpool: Starting new HTTPS connection (1): slack.com
2017-09-23 12:41:56 INFO root: bot is exiting
2017-09-23 12:41:56 INFO plugins: monkeypatch.otr_support unloaded
2017-09-23 12:41:56 INFO plugins: commands.plugincontrol unloaded
2017-09-23 12:41:56 INFO plugins: commands.basic unloaded
2017-09-23 12:41:56 INFO plugins: commands.tagging unloaded
2017-09-23 12:41:56 INFO plugins: commands.permamem unloaded
2017-09-23 12:41:56 INFO plugins: commands.convid unloaded
2017-09-23 12:41:56 INFO plugins: commands.loggertochat unloaded
2017-09-23 12:41:56 INFO plugins: plugins.botaliases unloaded
2017-09-23 12:41:56 INFO plugins: plugins.default unloaded
2017-09-23 12:41:56 INFO plugins: plugins.mentions unloaded
2017-09-23 12:41:56 INFO plugins: plugins.subscribe unloaded
2017-09-23 12:41:56 INFO plugins: plugins.autoreply unloaded
2017-09-23 12:41:56 INFO plugins: plugins.dnd unloaded
2017-09-23 12:41:56 INFO plugins: plugins.starter unloaded
2017-09-23 12:41:56 INFO plugins: plugins.remind unloaded
2017-09-23 12:41:56 INFO plugins: plugins.xkcd unloaded
2017-09-23 12:41:56 INFO plugins: plugins.catfact unloaded
2017-09-23 12:41:56 INFO plugins: plugins.slackrtm unloaded
```

**Comment by das7pad** [comment]
As noted in your PR: I suggest taking a look at the [v3 branch](https://github.com/hangoutsbot/hangoutsbot/tree/v3.0/hangupsbot/plugins/slackrtm). v2 uses an outdated version of `hangups`. I guess the reference to an attached reprocessor is not sent anymore with the old-version.

**Comment by MarkNBroadhead** [comment]
Is work being done on v3? I see it marked as beta and no PRs have been merged in months...

**Comment by MTRNord** [comment]
@MarkNBroadhead v3 is way more stable than v2. I run a sync bot since end last year and currently have no issues.

**Comment by OllieTerrance** [comment]
Dev progress is currently rather light as a whole, but no more work is going into v2 in preparation for making v3 the default -- it should be stable enough for everyday use.  I would suggest trying the upgrade if you haven't already, it shouldn't require any config changes (besides loading the `image` plugin if you use other syncs).

Notably, SlackRTM has seen significant changes which may resolve the original issue here, both in the plugin itself and the underlaying bridge framework.  There's also another (ongoing) rewrite of SlackRTM in #833 which aims to clean up a lot of cruft in the current plugin.

> Please note lines 50-53. It is having an issue parsing image data. My bot does not have an image for Hangouts.

This is #336, which was fixed in commit 96363f7 by surrounding the photo URL access in a try-catch (which emits an error, but otherwise ignores it and continues as normal).

**Comment by Terrance** [comment]
v3 has now been promoted to default, which includes the SlackRTM rewrite to cover the issues in here.
